go/extractor/extractor.go

```diff
@@ -1507,9 +1507,24 @@ func extractSpec(tw *trap.Writer, spec ast.Spec, parent trap.Label, idx int) {
 	extractNodeLocation(tw, spec, lbl)
 }
 
+// Determines whether the given type is an alias.
+func isAlias(tp types.Type) bool {
+	_, ok := tp.(*types.Alias)
+	return ok
+}
+
+// If the given type is a type alias, this function resolves it to its underlying type.
+func resolveTypeAlias(tp types.Type) types.Type {
+	if isAlias(tp) {
+		return types.Unalias(tp) // tp.Underlying()
+	}
+	return tp
+}
+
```

Comment:
```suggestion
```
I think this is all redundant and we can just use `types.Unalias`. But this can be done in a follow-up PR.